client/state/posts/test/actions.js

```diff
@@ -153,4 +163,70 @@ describe( 'actions', () => {
 			} ).catch( done );
 		} );
 	} );
+
+	describe( '#requestSitePost()', () => {
+		before( () => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
+				.get( '/rest/v1.1/sites/2916284/posts/413' )
+				.reply( 200, { ID: 413, title: 'Ribs & Chicken' } )
+				.get( '/rest/v1.1/sites/2916284/posts/420' )
+				.reply( 404, {
+					error: 'unknown_post',
+					message: 'Unknown post'
+				} );
+		} );
+
+		after( () => {
+			nock.cleanAll();
+		} );
+
+		it( 'should dispatch request action when thunk triggered', () => {
+			requestSitePost( 2916284, 413 )( spy );
+
+			expect( spy ).to.have.been.calledWith( {
+				type: POST_REQUEST,
+				siteId: 2916284,
+				postId: 413
+			} );
+		} );
+
+		it( 'should dispatch posts receive action when request completes', ( done ) => {
+			requestSitePost( 2916284, 413 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: POSTS_RECEIVE,
+					posts: [
+						sinon.match( { ID: 413, title: 'Ribs & Chicken' } )
+					]
+				} );
+
+				done();
+			} ).catch( done );
+		} );
+
+		it( 'should dispatch posts posts request success action when request completes', ( done ) => {
+			requestSitePost( 2916284, 413 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: POST_REQUEST_SUCCESS,
+					siteId: 2916284,
+					postId: 413
+				} );
+
+				done();
+			} ).catch( done );
+		} );
+
+		it( 'should dispatch fail action when request fails', ( done ) => {
+			requestSitePost( 2916284, 420 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: POST_REQUEST_FAILURE,
+					siteId: 2916284,
+					postId: 420,
+					error: sinon.match( { message: 'Unknown post' } )
+				} );
+
+				done();
+			} ).catch( done );
```

Comment:
>Do we want the Promise to resolve or reject if the API request bombs?

We want the promise to resolve successfully regardless of whether an API failure occurs. API failures are handled by the `POST_REQUEST_FAILURE` action.